{"package_name": "java.security", "subclass_list": [], "class_description": "Legacy security code; do not use.\n\n This exception is thrown by\n doPrivileged(PrivilegedExceptionAction) and\n doPrivileged(PrivilegedExceptionAction,\n AccessControlContext context) to indicate\n that the action being performed threw a checked exception.  The exception\n thrown by the action can be obtained by calling the\n getException method.  In effect, an\n PrivilegedActionException is a \"wrapper\"\n for an exception thrown by a privileged action. As of release 1.4, this exception has been retrofitted to conform to\n the general purpose exception-chaining mechanism.  The \"exception thrown\n by the privileged computation\" that is provided at construction time and\n accessed via the getException() method is now known as the\n cause, and may be accessed via the Throwable#getCause()\n method, as well as the aforementioned \"legacy method.\" ", "class_type": "class", "typealias": [], "interface_list": [""], "class_inherit_list": ["Exception"], "Methods": [{"params": [{"param_name": ["exception"], "param_type": "Exception", "param_description": " The exception thrown"}], "class_name": "PrivilegedActionException", "is_static": false, "return_value": {"return_name": "", "return_description": [], "return_type": []}, "method_name": "init", "method_description": "Constructs a new PrivilegedActionException \"wrapping\"\n the specific Exception.\n", "method_declar": ""}, {"params": [], "class_name": "PrivilegedActionException", "is_static": false, "return_value": {"return_name": [], "return_description": ["the cause of this exception."], "return_type": ["Throwable"]}, "method_name": "getCause", "method_description": "Returns the cause of this exception (the exception thrown by\n the privileged computation that resulted in this\n PrivilegedActionException).\n", "method_declar": ""}, {"params": [], "class_name": "PrivilegedActionException", "is_static": false, "return_value": {"return_name": [], "return_description": ["the exception thrown by the privileged computation that\n         resulted in this PrivilegedActionException."], "return_type": ["Exception"]}, "method_name": "getException", "method_description": "Returns the exception thrown by the privileged computation that\n resulted in this PrivilegedActionException.\n\n \nThis method predates the general-purpose exception chaining facility.\n The Throwable#getCause() method is now the preferred means of\n obtaining this information.\nSee also:\n", "method_declar": ""}, {"params": [], "class_name": "PrivilegedActionException", "is_static": false, "return_value": {"return_name": [], "return_description": ["a string representation of this throwable."], "return_type": ["String"]}, "method_name": "toString", "method_description": "Returns a short description of this throwable.\n The result is the concatenation of:\n \n", "method_declar": ""}], "class_name": "PrivilegedActionException", "Vars": null}